autobotAI_integrations/utils/aws_mcp_auth.py

```
@@ -0,0 +1,187 @@
+
+    creds = build_aws_mcp_creds(
+        access_key_id=access_key_id,
+        secret_access_key=secret_access_key,
+        session_token=session_token,
+        sigv4_region=region,
+        default_aws_region=default_aws_region,
+    )
+    sigv4_service = creds.aws_sigv4_service or AWS_MCP_SIGV4_SERVICE
+
+    class _AWSMCPSigV4Auth(httpx.Auth):
+        def auth_flow(self, request):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `auth_flow`

(ANN202)